### PR TITLE
[ROCKETMQ-219] Add batch example

### PR DESCRIPTION
**Pull request by dongeforever**
https://issues.apache.org/jira/browse/ROCKETMQ-219

@lizhanhui @zhouxinyu

### CODE REVIEW

File: example/src/main/java/org/apache/rocketmq/example/batch/SplitBatchProducer.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+package org.apache.rocketmq.example.batch;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.rocketmq.client.producer.DefaultMQProducer;
+import org.apache.rocketmq.common.message.Message;
+
+public class SplitBatchProducer {
```
Comment (lizhanhui) [inline example/src/main/java/org/apache/rocketmq/example/batch/SplitBatchProducer.java]: Good to add a concise explanation why we should split when sending in batch. That is,  total message size of messages in a batch cannot exceed 1 mega bytes on default.

**Comment by lizhanhui** [comment]
Overall +1

**Comment by zhouxinyu** [comment]
+1, LGTM~

**Comment by vongosling** [comment]
+1

**Comment by dongeforever** [comment]
@zhouxinyu Thanks

**Comment by ranqiqiang** [comment]
rocketmq version 4.5  ,  SimpleBatchProducer  会出现：
“the message is illegal, maybe msg body or properties length not matched. msg body length limit 128k, msg properties length limit 32k.”
看demo  数据量很小的啊，版本4.3  就不会出现，是broker 什么设置不对吗？

**Pull request closed without merging**